client/app/components/QuerySelector.jsx

```diff
@@ -116,6 +119,30 @@ export function QuerySelector(props) {
     return <Input value={selectedQuery && selectedQuery.name} placeholder={placeholder} disabled />;
   }
 
+  if (props.type === 'select') {
+    const suffixIcon = selectedQuery ? clearIcon : null;
+    const value = selectedQuery ? selectedQuery.name : searchTerm;
+
+    return (
+      <Select
+        showSearch
+        dropdownMatchSelectWidth={false}
+        placeholder={placeholder}
+        value={value || undefined} // undefined for the placeholder to show
+        onSearch={setSearchTerm}
+        onChange={selectQuery}
+        suffixIcon={searching ? spinIcon : suffixIcon}
+        notFoundContent={null}
+        filterOption={false}
+        defaultActiveFirstOption={false}
+      >
+        {searchResults && searchResults.map(q => (
+          <Option value={q.id} key={q.id}>{q.name}</Option>
```

Comment:
Can option render "richer" content?

Reply:
@arikfr Yes, it can render anything

Reply:
_(even `<iframe>` I think :smile: )_

Reply:
I'll try the tags

Reply:
Then we probably should show the same info as in the non `select` version?

Reply:
Also in Add widget dialog there was a feature (IIRC) when it shows unpublished queries too, but does not allow to select them - I think `<Option disabled>` is perfect for this.

Reply:
Yeah I missed that completely. On it.

Reply:
Fixed.
![screen shot 2019-02-27 at 12 44 02](https://user-images.githubusercontent.com/486954/53484912-9a988f80-3a8d-11e9-98ff-eec0e28f4cd5.png)

Reply:
Looks really nice :+1: